src/Illuminate/Bus/DynamoBatchRepository.php

```diff
@@ -0,0 +1,484 @@
+     */
+    public function markAsFinished(string $batchId)
+    {
+        $update = 'SET finished_at = :timestamp';
+        if ($this->ttl !== null) {
+            $update = "SET finished_at = :timestamp, #{$this->ttlAttribute} = :ttl";
+        }
+        $this->dynamoDbClient->updateItem(array_filter([
+            'TableName' => $this->table,
+            'Key' => [
+                'application' => ['S' => $this->applicationName],
+                'id' => ['S' => $batchId],
+            ],
+            'UpdateExpression' => $update,
+            'ExpressionAttributeValues' => array_filter([
+                ':timestamp' => ['N' => (string) time()],
+                ':ttl' => array_filter(['N' => $this->getExpiryTime()]),
+            ]),
+            'ExpressionAttributeNames' => $this->ttlExpressionAttributeName(),
+        ]));
+    }
+
+    protected function getExpiryTime(): ?string
+    {
+        if ($this->ttl === null) {
+            return null;
+        }
+
+        return (string) (time() + $this->ttl);
+    }
+
+    protected function ttlExpressionAttributeName(): array
+    {
+        if ($this->ttl === null) {
+            return [];
+        }
+
+        return ["#{$this->ttlAttribute}" => $this->ttlAttribute];
+    }
+
+    /**
+     * Cancel the batch that has the given ID.
+     *
+     * @param  string  $batchId
+     * @return void
+     */
+    public function cancel(string $batchId)
+    {
+        $update = 'SET cancelled_at = :timestamp, finished_at = :timestamp';
+        if ($this->ttl !== null) {
+            $update = "SET cancelled_at = :timestamp, finished_at = :timestamp, #{$this->ttlAttribute} = :ttl";
+        }
+
+        $this->dynamoDbClient->updateItem(array_filter([
+            'TableName' => $this->table,
+            'Key' => [
+                'application' => ['S' => $this->applicationName],
+                'id' => ['S' => $batchId],
+            ],
+            'UpdateExpression' => $update,
+            'ExpressionAttributeValues' => array_filter([
+                ':timestamp' => ['N' => (string) time()],
+                ':ttl' => array_filter(['N' => $this->getExpiryTime()]),
+            ]),
+            'ExpressionAttributeNames' => $this->ttlExpressionAttributeName(),
+        ]));
+    }
+
+    /**
+     * Delete the batch that has the given ID.
+     *
+     * @param  string  $batchId
+     * @return void
+     */
+    public function delete(string $batchId)
+    {
+        $this->dynamoDbClient->deleteItem([
+            'TableName' => $this->table,
+            'Key' => [
+                'application' => ['S' => $this->applicationName],
+                'id' => ['S' => $batchId],
+            ],
+        ]);
+    }
+
+    /**
+     * Execute the given Closure within a storage specific transaction.
+     *
+     * @param  \Closure  $callback
+     * @return mixed
+     */
+    public function transaction(Closure $callback)
+    {
+        return $callback();
+    }
```

Comment:
as a forward compatibility can you also add the `rollBack()` method (does nothing) as this will be enforced in Laravel 11.

Reply:
Done